test/src/us/kbase/kbasefeaturevalues/test/KBaseFeatureValuesServerTest.java

```diff
@@ -329,38 +356,87 @@ public void testPyScikitKMeans() throws Exception {
         String clustObj1Name = "py_clusters1";
         ExpressionMatrix data = new ExpressionMatrix().withType("log-ratio").withScale("1.0")
                 .withData(getSampleMatrix());
-        WorkspaceClient wscl = getWsClient();
-        wscl.saveObjects(new SaveObjectsParams().withWorkspace(testWsName).withObjects(Arrays.asList(
+        getWsClient().saveObjects(new SaveObjectsParams().withWorkspace(testWsName).withObjects(Arrays.asList(
                 new ObjectSaveData().withName(exprObjName).withType("KBaseFeatureValues.ExpressionMatrix")
                 .withData(new UObject(data)))));
-        impl.clusterKMeans(new ClusterKMeansParams().withInputData(testWsName + "/" + 
+        String getId = impl.clusterKMeans(new ClusterKMeansParams().withInputData(testWsName + "/" + 
                 exprObjName).withK(3L).withOutWorkspace(testWsName).withOutClustersetId(clustObj1Name),
                 token, getContext());
-        ObjectData res = wscl.getObjects(Arrays.asList(new ObjectIdentity().withWorkspace(testWsName)
+        ObjectData res = getWsClient().getObjects(Arrays.asList(new ObjectIdentity().withWorkspace(testWsName)
                 .withName(clustObj1Name))).get(0);
+
         FeatureClusters clSet = res.getData().asClassInstance(FeatureClusters.class);
         checkKMeansForSample(clSet);
     }
-    
+
+    @Test
```

Comment:
I don't understand the point of this test at all. It's just testing the workspace, not feature values.

Reply:
I think there's some misunderstanding here about testing, so before I spend more time on the review I think we should have a bit of a natter and get on the same page, to use a ghastly business idiom.